### PR TITLE
Fix: Custom Width on tablet/mobile generates wrong value when desktop is set to default [ED-8122]

### LINKED ISSUE #19542: 🐞 Bug report: important bug in elementor 3.7 - section
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Before updating to 3.7, I had a section and added 4 icon widgets inside it. I had considered the width of each 25%. **(The important bit here is that I entered a width of 25% for tablet mode so that it will be inherited on mobile. But it was the default desktop)**

### Steps to reproduce

1- Add a section and add 4 icon widgets inside it.
2- To determine the width of the icons, enter the tablet mode and consider the width of each icon to be 25%.
3- Now view the website with your mobile phone. 25% width is ignored and icons are compressed to one side.

**some notes**
A: If we enter the width of icon widgets first in desktop mode, this bug does not occur. But if we enter it first in the tablet, the width is ignored in the mobile and the icons are compressed on one side.
B : Everything is fine in the editor. But if you view the website on mobile, the situation is not good.

Please watch the video below : 

https://user-images.githubusercontent.com/77163248/185743746-4d07dd02-ed76-4cf9-aed4-0f176822bb74.mp4

@shilo-ey , Please let me know if you are not successful in reproducing. I tested this bug on 2 websites.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

-

### PR DESCRIPTION
**Pull request by udidol**
## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Fix: Custom Width on tablet/mobile generates wrong value when desktop is set to default
* Fix: Conditions for responsive controls worked in the Editor but not in the frontend in some cases

## Description
An explanation of what is done in this PR

* Fixes #19542, #19528
* If a responsive control was conditioned upon a responsive control (Like in `Advanced > Layout > Width`), the conditioning was respected in the Editor, allowing you to edit the responsive values of the conditioned control when the condition was met. However, in such cases, non-desktop values did not get applied in the front end's CSS. That was the bug.

### CODE REVIEW

**Review by TzviRabinovitch — CHANGES_REQUESTED** [review]
tests?

File: tests/phpunit/elementor/includes/base/test-controls-stack.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+		],
+		'elements' => [],
+		'widgetType' => 'button',
+	];
+
+	public function test_is_control_visible_when_condition_met() {
+		$initial_duplication_mode = Plugin::$instance->breakpoints->get_responsive_control_duplication_mode();
+
+		/**
+		 * Arrange
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/includes/base/test-controls-stack.php:32]: duplicate comment

**Review by TzviRabinovitch — DISMISSED** [review]

**Review by TzviRabinovitch — APPROVED** [review]

**Comment by Michaelgimii** [comment]
@udidol , Are the following things addressed in this amendment? (as a reminder)

- Custom position & Size in background : https://prnt.sc/gw-mQJqGcFOR
- Custom position & Size in background overlay
- The above cases occur even in the background widgets. https://prnt.sc/xTlXFXBeWwb5

**Comment by udidol** [comment]
@Michaelgimii All of these cases should be fixed by this ammendment.

**Pull request merged**